src/Sets/Zonotope.jl

```diff
@@ -371,19 +371,128 @@ nonlinear systems with uncertain parameters using conservative linearization.
+         corresponding generator
+
+### Output
+
+The zonotopes obtained by splitting `Z` into `2^{n_i}` zonotopes for each
+generator `i` such that their union is `Z` and their intersection is
+possibly non-empty.
+
+### Examples
+
+Splitting of a two-dimensional zonotope along its first generator:
+
+```jldoctest zonotope_label
+julia> Z = Zonotope([1.0, 0.0], [0.1 0.0; 0.0 0.1])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.0, 0.0], [0.1 0.0; 0.0 0.1])
+
+julia> split(Z, [1], [1])
+2-element Array{Zonotope{Float64,Array{Float64,1},Array{Float64,2}},1}:
+ Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.95, 0.0], [0.05 0.0; 0.0 0.1])
+ Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.05, 0.0], [0.05 0.0; 0.0 0.1])
+```
+Here, the first vector in the arguments corresponds to the zonotope's
+generator to be split, and the second vector corresponds to the exponent of
+`2^n` times that the zonotope will be split along the corresponding generator.
+
+Splitting of a two-dimensional zonotope along its generators:
+
+```
+julia> Z = Zonotope([1.0, 0.0], [0.1 0.0; 0.0 0.1])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.0, 0.0], [0.1 0.0; 0.0 0.1])
+
+julia> split(Z, [1, 2], [2, 2])
+16-element Array{Zonotope{Float64,Array{Float64,1},Array{Float64,2}},1}:
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.925, 0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([0.975, 0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.025, 0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, -0.075], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, -0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, 0.025], [0.025 0.0; 0.0 0.025])
+Zonotope{Float64,Array{Float64,1},Array{Float64,2}}([1.075, 0.075], [0.025 0.0; 0.0 0.025])
+```
+Here the zonotope is split along both of its generators four times.
+"""
+function split(Z::Zonotope, gens::AbstractVector, n::AbstractVector)
```

Comment:
I think the code is general for `AbstractZonotope`.

Reply:
yes indeed, that should be generalized, as well as the code for `split(Z, j)`.

Reply:
i didn't pretend working further in this PR, but it's fine either way.

Reply:
We can keep the other `split` specific but here it is a matter of changing the signature only.

Reply:
is there a particular reason to keep the `split(Z, j)` specific for Zonotopes? `split(Z::Zonotope, gens::AbstractVector, n::AbstractVector)` is a generalization.

Reply:
You sounded like you want to merge this one soon.

Reply:
the code has got in much better shape after each iteration, it's not bad to do it once again :D